api/src/utils/onboarding.ts

```diff
@@ -0,0 +1,62 @@
+import type { OnboardingPayload, UserOnboarding } from '@directus/types';
+import logger from '../logger.js';
+import { SettingsService } from '../services/settings.js';
+import { UsersService } from '../services/users.js';
+import { getSchema } from './get-schema.js';
+
+export async function collectOnboarding(userId: string) {
+	const axios = (await import('axios')).default;
+
+	const schema = await getSchema();
+	const usersService = new UsersService({ schema });
+	const user = await usersService.readOne(userId, { fields: ['email', 'onboarding'] });
+
+	// Some databases that dont have a native json type may return strings
+	if (typeof user?.['onboarding'] === 'string') {
+		user['onboarding'] = JSON.parse(user['onboarding']);
+	}
+
+	if (!user?.['onboarding']?.['retryTransmission']) {
+		return;
+	}
+
+	const settingsService = new SettingsService({ schema });
+
+	const settings = await settingsService.readSingleton({
+		fields: ['project_name', 'project_url', 'onboarding'],
+	});
+
+	if (typeof settings?.['onboarding'] === 'string') {
+		settings['onboarding'] = JSON.parse(settings['onboarding']);
```

Comment:
This could probably use a try catch in case the string does not turn out to be a valid JSON string. Or perhaps use the `parseJSON` utility